configs/projects/_shared-agent-settings.rb

```diff
@@ -156,10 +156,13 @@
   proj.setting(:ldflags, "-L#{proj.libdir} -Wl,-rpath=#{proj.libdir},-z,relro,-z,now")
 end
 
-if platform.name =~ /^redhatfips-/
+case platform.name
+when /^redhatfips-/
   proj.setting(:openssl_version, '1.1.1-fips')
-elsif platform.name =~ /^windowsfips-2012r2/
+when /^windowsfips-2012r2/
   proj.setting(:openssl_version, '1.0.2')
+when /^el-/, /^redhat-/, /^fedora-/, /^debian-/, /^ubuntu-/, /^sles-/
```

Comment:
Do we intend to land openssl 3.0 for the agent-runtime-7.x/6.x?

Reply:
ah good catch, no just in `main`. I'll need to override the `openssl_version` in `agent-runtime-main`